rmw_connext_cpp/src/functions.cpp

```diff
@@ -132,61 +132,87 @@ rmw_create_publisher(
   const char * topic_name,
   size_t queue_size)
 {
+  if (!node) {
+    rmw_set_error_string("node handle is null");
+    return NULL;
+  }
   if (node->implementation_identifier != rti_connext_identifier) {
-    rmw_set_error_string("node handle not from this implementation");
-    // printf("but from: %s\n", node->implementation_identifier);
+    rmw_set_error_string("node handle is not from this rmw implementation");
+    return NULL;
+  }
+  if (!type_support) {
+    rmw_set_error_string("type support handle is null");
     return NULL;
   }
-
-  DDSDomainParticipant * participant = static_cast<DDSDomainParticipant *>(node->data);
-
   if (type_support->typesupport_identifier !=
     rosidl_typesupport_connext_cpp::typesupport_connext_identifier)
   {
-    rmw_set_error_string("type support not from this implementation");
-    // printf("but from: %s\n", type_support->typesupport_identifier);
+    rmw_set_error_string("type support is not from this rmw implementation");
+    return NULL;
+  }
+
+  DDSDomainParticipant * participant = static_cast<DDSDomainParticipant *>(node->data);
+  if (!participant) {
+    rmw_set_error_string("participant handle is null");
     return NULL;
   }
 
-  const message_type_support_callbacks_t * callbacks = \
+  const message_type_support_callbacks_t * callbacks =
     static_cast<const message_type_support_callbacks_t *>(type_support->data);
+  if (!callbacks) {
+    rmw_set_error_string("callbacks handle is null");
+    return NULL;
+  }
   std::string type_name = std::string(callbacks->package_name) + "::msg::dds_::" +
     callbacks->message_name + "_";
 
-  callbacks->register_type(participant, type_name.c_str());
+  bool registered = callbacks->register_type(participant, type_name.c_str());
+  if (!registered) {
+    rmw_set_error_string("failed to register type");
+    return NULL;
+  }
 
 
   DDS_PublisherQos publisher_qos;
   DDS_ReturnCode_t status = participant->get_default_publisher_qos(publisher_qos);
   if (status != DDS_RETCODE_OK) {
     rmw_set_error_string("failed to get default publisher qos");
-    // printf("get_default_publisher_qos() failed. Status = %d\n", status);
     return NULL;
   }
 
   DDSPublisher * dds_publisher = participant->create_publisher(
     publisher_qos, NULL, DDS_STATUS_MASK_NONE);
   if (!dds_publisher) {
-    rmw_set_error_string("  create_publisher() could not create publisher");
+    rmw_set_error_string("failed to create publisher");
     return NULL;
   }
 
 
-  DDS_TopicQos default_topic_qos;
-  status = participant->get_default_topic_qos(default_topic_qos);
-  if (status != DDS_RETCODE_OK) {
-    rmw_set_error_string("failed to get default topic qos");
-    // printf("get_default_topic_qos() failed. Status = %d\n", status);
-    return NULL;
-  }
+  DDSTopic * topic;
+  DDSTopicDescription * topic_description = participant->lookup_topicdescription(topic_name);
```

Comment:
This is necessary to fix ros2/system_tests#13. Otherwise the second time a topic is created results in a crash of `rmw_connext_*`.